backend/services/implementations/schoolService.ts

```diff
@@ -52,6 +52,26 @@ class SchoolService implements ISchoolService {
     }
   }
 
+  /**
+   * This method gets all schools with the given country from the database.
+   */
+  async getSchoolByCountry(country: string): Promise<SchoolResponseDTO[]> {
+    let schools: Array<School> | null;
+
+    try {
+      schools = await MgSchool.find({ country });
+
+      if (!schools.length) {
+        throw new Error(`Country ${country} not found`);
+      }
```

Comment:
@joyce-shi - should we be throwing errors here (and in the other services), if we don't find anything? I feel like we could just return an empty array here and the upstream caller can do whatever it wants with it.

Reply:
+1 to that. I think the most likely use case for this is that admin staff would be selecting from a pre-populated dropdown to query by country. Given this, no error throwing necessary.